editor/src/core/model/element-metadata-utils.ts

```diff
@@ -532,21 +547,25 @@ export const MetadataUtils = {
     })
     return jsxMetadata(metadata.components, elements)
   },
-  getImmediateChildrenPaths(metadata: JSXMetadata, target: TemplatePath): Array<InstancePath> {
-    if (TP.isScenePath(target)) {
-      const scene = MetadataUtils.findSceneByTemplatePath(metadata.components, target)
-      return scene?.rootElements ?? []
-    } else {
-      const element = MetadataUtils.getElementByInstancePathMaybe(metadata.elements, target)
-      return element?.children ?? []
-    }
+  getRootViews(elements: ElementInstanceMetadataMap, target: TemplatePath): Array<InstancePath> {
+    const element = MetadataUtils.findElementByTemplatePath(elements, target)
+    return element?.rootElements ?? []
+  },
+  getImmediateChildrenPaths(
+    elements: ElementInstanceMetadataMap,
+    target: TemplatePath,
+  ): Array<InstancePath> {
+    const element = MetadataUtils.findElementByTemplatePath(elements, target)
+    const rootElements = element?.rootElements ?? []
+    const children = element?.children ?? []
+    return [...rootElements, ...children]
```

Comment:
Surely return the merged arrays if `element` is not null and return an empty one otherwise?

Reply:
Yeah that's fair :+1: